builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,340 @@
+
+1. Click on the **Compile** tab, second from top
+2. Then to compile the contract, click on **Compile WhitelistFunctionSelector.sol**
+
+![Compiling WhitelistFunctionSelector.sol](/images/builders/get-started/eth-compare/security/security-5.png)
+
+Then take the following steps to deploy the contract on Moonbase Alpha:
+
+1. Click on the **Deploy and Run** tab, directly below the **Compile** tab in Remix
+2. Make sure **Injected Provider - Metamask** is selected in the **ENVIRONMENT** drop down and that you're connected to Moonbase Alpha in MetaMask
+3. Ensure **WhitelistFunctionSelector - WhitelistFunctionSelector.sol** is selected in the **CONTRACT** dropdown
+4. Click **Deploy** and confirm the transaction in MetaMask
+
+![Deploy the contract](/images/builders/get-started/eth-compare/security/security-6.png)
+
+Once the transaction goes through, the `WhitelistFunctionSelector.sol` contract will appear in the list of **Deployed Contracts**.
+
+You can repeat these steps to compile and deploy the `SetMessage.sol` contract.
+
+### Execute the Whitelisted Function {: #execute-the-whitelisted-function }
+
+Now that you have the example contracts deployed, you can verify that you can execute the whitelisted function only. To do so, you can take the following steps:
+
+1. Copy the address of the **SETMESSAGE** contract
+2. Expand the **WHITELISTFUNCTIONSELECTOR** contract
+3. Expand the **whitelistFunctionSelector** function
+4. Paste the address of the `SetMessage.sol` contract in the **_target** field
+5. For the **_bytes** field, you can enter the following encoded call data which will set the message to `hello`: 
+    ```
+    0x4ed3885e0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000568656c6c6f000000000000000000000000000000000000000000000000000000
+    ```
+6. Click on **transact** and confirm the transaction in MetaMask
+
+![Execute the whitelisted function](/images/builders/get-started/eth-compare/security/security-7.png)
+
+Once the transaction goes through, you can then verify that the message has been set to `hello` in the `SetMessage.sol` contract:
+
+1. Expand the **SETMESSAGE** contract
+2. Click on **get**
+3. The response will appear right below the **get** button, which should be `hello`
+
+![Verify the whitelisted function was executed](/images/builders/get-started/eth-compare/security/security-8.png)
+
+To make sure that you cannot call any other function, you can repeat the previous steps to call the `whitelistFunctionSelector` function by taking the following steps:
+
+1. Expand the **whitelistFunctionSelector** function
+2. In the **_target** field, enter the contract address of the `SetMessage.sol` contract
+3. In the **_bytes** field, you can enter any other ABI-encoded call data. For example, you can use the bytes from the previous example that called 
```

Comment:
It might be useful to include an example of this in the same contract. For example:

```
contract SetMessage {
    string storedMessage;
    string nonWhitelistedMessage;

    function set(string calldata x) public {
        storedMessage = x;
    }

    function nonWhitelistedSet(string calldata x) public {
        nonWhitelistedMessage = x;
    }

    function get() public view returns (string memory) {
        return storedMessage;
    }
}
```